### PR TITLE
fix: fixed exporting problems with meta data

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
We right now have the problem that metadata contains classes from editor namespace which is not put into built. To fix this I added a post built hook which removes all metadata from courses inside the built project.

**Fixes**: # (issue)

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

- Manually looking into the code, seeing if metadata was removed.

### CODE REVIEW

File: Editor/OnBuildMetaDataRemoval.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+namespace Innoactive.CreatorEditor
+{
+    internal class OnBuildMetaDataRemoval
+    {
+        [PostProcessBuild()]
+        public static void OnPostprocessBuild(BuildTarget target, string pathToBuiltProject)
+        {
+            if (EditorConfigurator.Instance.Serializer is IOnBuildExportSerialization serializer)
+            {
+                string trainingFolder = Path.Combine(Path.GetDirectoryName(pathToBuiltProject), "Project_Data/StreamingAssets/Training");
```
Comment (Gusinuhe) [inline Editor/OnBuildMetaDataRemoval.cs:21]: This only works on Windows.

**Comment by aleksei-korolev** [comment]
What editor classes get serialized?

**Comment by SimonTheSourcerer** [comment]
it's only one ReorderableElementMetadata

**Comment by SimonTheSourcerer** [comment]
But it does also force us to put a lot of stuff utterly unrelated to runtime into runtime.

**Comment by aleksei-korolev** [comment]
I would argue that it would be less error-prone if we do it the other way around, by deserializing only what we need, so we wouldn't have to keep track of two different versions of the course. We can do it either with json converter for Metadata type, or we could introduce a new attribute [DeserializeOnlyInEditor]

**Comment by SimonTheSourcerer** [comment]
Does not work, because converter are working with the Type, which cannot be made out of thin air.

**Comment by SimonTheSourcerer** [comment]
This probably might help, trying it tomorrow:
https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_Serialization_ISerializationBinder.htm

**Comment by SimonTheSourcerer** [comment]
Tried multiple approaches to extract only course data from newtonsoft on import. Non worked for a ton of different reasons. As long as we store metadata the way we do, i dont see way to remove them on load :(

**Pull request closed without merging**